testutils/clustermanager/gke.go

```diff
@@ -260,13 +261,70 @@ func (gc *GKECluster) Acquire() error {
 	return err
 }
 
-// Delete deletes a GKE cluster
-func (gc *GKECluster) Delete() error {
-	if !gc.NeedCleanup {
-		return nil
+// wait depends on unique opName(operation ID created by cloud), and waits until
+// it's done
+func (gc *GKECluster) wait(location, opName string, wait time.Duration) error {
+	validStatuses := []string{"PENDING", "RUNNING", "DONE"}
+	doneStatus := "DONE"
+	var op *container.Operation
+	var err error
+
+	timeout := time.After(wait)
+	tick := time.Tick(500 * time.Millisecond)
+	for {
+		select {
+		// Got a timeout! fail with a timeout error
+		case <-timeout:
+			return fmt.Errorf("timed out waiting")
+		case <-tick:
+			// Retry 3 times in case of weird network error, or rate limiting
+			r := 0
+			for w := 50 * time.Microsecond; r < 3; w *= 2 {
```

Comment:
`for r, w := 0, 50; r < 3; r, w := r+1,w*2 {}`

Reply:
seems like python, good to learn. Done